lib/ts/recipe/emailverification/api/implementation.ts

```diff
@@ -16,6 +18,8 @@ export default function getAPIInterface(): APIInterface {
         }): Promise<
             { status: "OK"; user: User } | { status: "EMAIL_VERIFICATION_INVALID_TOKEN_ERROR" } | GeneralErrorResponse
         > {
+            // TODO: we could set isEmailVerified here if we have a session with the right user
```

Comment:
We should.

In the case that we don't have it, then how would the user experience be?

Reply:
They should get redirected to the auth page.